metagraph-plugin/metagraph_katana/plugins/metagraph_katana/translators.py

```diff
@@ -48,12 +48,24 @@ def networkx_to_katanagraph(x: NetworkXGraph, **props) -> KatanaGraph:
 
 @translator
 def katanagraph_to_networkx(x: KatanaGraph, **props) -> NetworkXGraph:
+    pg = x.value
+    dest_list = [dest for src in pg for dest in [pg.get_edge_dest(e) for e in pg.edges(src)] ]
```

Comment:
I don't think you need to build this list. I think you can generate the values in the loop below with no overhead. This will reduce memory usage a LOT on larger graphs.

Reply:
I do think I need a global dest_list generated before the for loop below. Because the purpose of this step is to capture the UN-ISOLATED nodes which serve as dest nodes as some edges, but not src nodes as some edges. For example, we have graph (1,2), (1,3), (2,0) (2,1), and the node 0 is actually not isolated, if we add dest_list generation inside the for loop below, the dest_list will be [], [2,3] [2,3,0,1] for the 3 iterations, so in the iteration 0, pg.edges(0)==range(0,0) is true, but we cannot find 0 in the dest_list.

```
    dest_list = [dest for src in pg for dest in [pg.get_edge_dest(e) for e in pg.edges(src)] ]
    for src in pg:
        if pg.edges(src) == range(0,0):
            assert src in dest_list, "NetworkX does not support graph with isolated nodes, please use a cleaned Katana Graph"
```

I am understanding this suggestion correct?

Or are there other elegant ways to find isolated nodes?

Reply:
You are right. I wasn't reading the code correctly. Carry on. :-)

Reply:
Got it.